src/application/look-up-wallet-address/look-up-wallet-address.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from 'react';
+import { useKey } from 'react-use';
+import { useForm } from 'react-hook-form';
+
+import { useCommandQuery } from 'shared/messaging';
+import { Closable, IconButton, Spinner } from 'shared/ui';
+import { useExtensionSettings } from 'shared/extension';
+
+import { GetResolvedAddressCommand, GetTwitterIdsCommand } from './commands';
+import { AddressList } from './address-list';
+
+export const LookUpWalletAddress = () => {
+  const [username, setUsername] = useState('');
+  const [isVisible, setIsVisible] = useState(false);
+
+  const { extensionSettings } = useExtensionSettings();
+
+  const isEnabled =
+    extensionSettings['entire-extension-enabled'] &&
+    extensionSettings['wallet-lookup-enabled'];
+
+  const twitterIdsQuery = useCommandQuery({
+    command: new GetTwitterIdsCommand({
+      username,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+  });
+
+  const addressesQuery = useCommandQuery({
+    command: new GetResolvedAddressCommand({
+      identifier: username,
+      twitterId: twitterIdsQuery.data?.twitterIDs
+        ? Object.values(twitterIdsQuery.data?.twitterIDs)[0]
+        : undefined,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+    enabled: username.length >= 3,
+  });
+
+  const form = useForm();
+
+  const userIdentifier = form.watch('user_identifier');
+
+  const onSubmit = () => {
+    const values = form.getValues();
+    setUsername(values.user_identifier);
+  };
+
+  const closeWidget = () => {
+    setIsVisible(false);
+    setUsername('');
+    form.reset();
+  };
+
+  useKey(
+    (event) => {
+      return event.ctrlKey && event.key === 'i';
+    },
+    () => {
+      return setIsVisible((previous) => {
+        return !previous;
+      });
```

Comment:
maybe we can move to toggle function wrapped with useCallback and do:
```ts
useKey(() => { return event.ctrlKey && event.key === 'i' }, toggleVisibility)
```

even better would be:
```tsx
// utils.ts
const isShortcut = (event: KeyboardEvent) => {
  return event.ctrlKey && event.key === 'i';
}

// component
useKey(isShortcut, toggleVisibility);
```